test/unit/host_test.rb

```diff
@@ -2174,6 +2174,14 @@ def to_managed!
     assert_equal 'foo', host.shortname
   end
 
+  test 'lookup_value_match returns host name instead of fqdn when there is no primary interface' do
+    host = FactoryGirl.build(:host, :managed)
```

Comment:
Adding `:name => 'mytesthost` was ignored by the factory that creates host names by sequence so I'm saving the host name on the next line.